src/Pyz/Yves/ProductDetailPage/Theme/default/components/molecules/variant/variant.twig

```diff
@@ -1,62 +1,39 @@
-{% extends model('component') %}
-{% import _self as component %}
-
-{% define config = {
-    name: 'variant',
-    tag: 'div'
-} %}
+{% extends molecule('variant', '@SprykerShop:ProductDetailPage') %}
 
 {% define data = {
-    name: required,
-    formName: required,
-    values: required,
-    selectedValue: null,
-    label: '',
-    optionResetUrl: '',
-    isAvailable: true,
     selectModifiers: ['margin-bottom'],
     selectTheme: 'default',
     selectPlaceholder: 'product.attribute._none',
-    autoInitSelect: false,
 } %}
 
 {% set selectModifiers = data.selectModifiers | merge(['full-width']) %}
 
 {% block body %}
-    <div class="{{ config.jsName }}__title {{ component.renderClass(config.name ~ '__title', modifiers) }}">{{data.label}}</div>
+    <div class="{{ _self.renderClass(config.name ~ '__title', modifiers) }}">{{ data.label }}</div>
     {% if data.selectedValue is null %}
         {% if data.isAvailable %}
-            {% embed molecule('custom-select') with {
+            {% set options = [{
+                label: data.selectPlaceholder | trans,
+                value: '',
+            }] %}
+
+            {% for value in data.values %}
+                {% set options = options | merge([{
+                    label: value,
+                    value: value,
+                }]) %}
+            {% endfor %}
+
+            {% include molecule('custom-select') with {
                 modifiers: selectModifiers,
-                class: 'name',
+                data: {
+                    options: options,
+                },
                 attributes: {
                     name: data.formName,
+                    'data-form-submitter': true,
                     'config-width': '100%',
                     'config-theme': data.selectTheme,
-                    'auto-init': data.autoInitSelect,
```

Comment:
Does molecule work correctly with this change?

Reply:
Yes, this attribute should be removed.
It already removed on [b2b demoshop](https://github.com/spryker/b2b-demo-shop-internal/pull/918)

Reply:
So please apply changes from the PR.